build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -2000,6 +2151,40 @@ message Compressor {
   }
 }
 
+// Content-defined chunking algorithms used for splitting blobs into chunks.
+message ChunkingAlgorithm {
+  enum Value {
+    // No chunking. Servers MUST always support this, and do not need to
+    // advertise it.
+    IDENTITY = 0;
+
+    // Content-defined chunking using Rabin fingerprints. An implementation of
+    // this scheme in presented in this paper
+    // https://link.springer.com/chapter/10.1007/978-1-4613-9323-8_11. The final
+    // implementation of this algorithm should be configured to have the
+    // following properties on resulting chunk sizes.
+    //  - Minimum chunk size: 2 KB
+    //  - Average chunk size: 8 KB (0x00000000007FFFFF bit mask)
+    //  - Maximum chunk size: 64 KB
+    // The irreducible polynomial to be used for the modulo divisions is the
+    // following 64-bit polynomial of degree 53: 0x003DA3358B4DC173. The window
+    // size to be used is 64 bits.
+    RABINCDC = 1;
+
+    // Content-defined chunking using the FastCDC algorithm. The algorithm is
+    // described in this paper https://ieeexplore.ieee.org/document/9055082
+    // (Algorithm 2, FastCDC8KB). The algorithm is configured to have the
+    // following properties on resulting chunk sizes.
+    //  - Minimum chunk size: 2 KB
+    //  - Average chunk size: 8 KB
```

Comment:
I understand that using small chunk sizes, such as 8 KB, can increase the likelihood of deduplication and may also reduce the risk of disk storage fragmentation. However, have you considered if there is potential performance overhead of having too many fine-grained CAS blobs?

I can envision that a feature like this could also be beneficial in distributing the load more evenly across multiple CAS shards. But for such use cases, it might make sense to use much larger chunks, perhaps 8 MB? Should we somehow accommodate also for larger chunks in this PR?

Reply:
We experimented with FastCDC on approximately 5TB of real Bazel data from many different codebases and found that 0.5MB is a good trade-off between space savings and metadata overhead. Too small of a chunk size means the metadata for all chunks becomes very large / numerous, while too large of a chunk size means poor space savings.

Reply:
Thanks @luluz66, I think 0.5 MB is more reasonable than 8 KB.

Do you think there is one value that would fit all, or should a size like this be allowed to be tuned in a flexible way?

Reply:
0.5 MB was the ideal range for us based on the Bazel-specific data set we were testing against. However, there would be no telling whether that number would be different for a different client/server pair, or a different data set.

So I think we would want a discovery mechanism for the FastCDC configuration on the server side. The client should follow the server's advertised setting in order to achieve the best result. WDYT?